File: WebSiteDatLich/src/main/java/com/example/WebSiteDatLich/model/Work_schedule.java
package com.example.WebSiteDatLich.model;

import java.util.List;

public class Work_schedule {
    private String work_schedule_id;
    private String schedule;
    private List<String> timeSlots; // Danh sách các khung giờ
    private String doctor_id;

    // Getter và Setter
    public String getWork_schedule_id() {
        return work_schedule_id;
    }

    public void setWork_schedule_id(String work_schedule_id) {
        this.work_schedule_id = work_schedule_id;
    }

    public String getSchedule() {
        return schedule;
    }

    public void setSchedule(String schedule) {
        this.schedule = schedule;
    }

    public List<String> getTimeSlots() {
        return timeSlots;
    }

    public void setTimeSlots(List<String> timeSlots) {
        this.timeSlots = timeSlots;
    }

    public String getDoctor_id() {
        return doctor_id;
    }

    public void setDoctor_id(String doctor_id) {
        this.doctor_id = doctor_id;
    }
}

File: ungdungdatlichkham/lib/models/WorkSchedule.dart
class WorkSchedule {
  dynamic workScheduleId; // Chấp nhận cả String và int
  String? schedule;
  List<String>? timeSlots;
  String? doctorId;

  WorkSchedule({
    this.workScheduleId,
    this.schedule,
    this.timeSlots,
    this.doctorId,
  });

  factory WorkSchedule.fromMap(Map<String, dynamic> map) {
    return WorkSchedule(
      workScheduleId: map['workScheduleId'], // Không cần chuyển đổi ở đây
      schedule: map['schedule'],
      timeSlots: map['timeSlots'] != null
          ? List<String>.from(map['timeSlots'] as List)
          : [],
      doctorId: map['doctor_id'],
    );
  }

  Map<String, dynamic> toMap() {
    return {
      'workScheduleId': workScheduleId,
      'schedule': schedule,
      'timeSlots': timeSlots,
      'doctor_id': doctorId,
    };
  }
}
